haskell/repl.bzl

```diff
@@ -124,7 +124,7 @@ def _create_HaskellReplCollectInfo(target, ctx):
             compiler_flags = getattr(ctx.rule.attr, "compiler_flags", []),
             repl_ghci_args = getattr(ctx.rule.attr, "repl_ghci_args", []),
         )
-    if HaskellLibraryInfo in target:
+    if HaskellLibraryInfo in target and CcInfo in target:
```

Comment:
This way the `package_id` and `package_databases` will be ignored if `CcInfo` is missing. You could instead use
```
cc_info = target[CcInfo] if CcInfo in target else cc_common.merge_cc_infos(cc_infos = [])
```
---
Which rule creates targets with `HaskellLibraryInfo` but without `CcInfo`? Maybe that rule should be extended to return `CcInfo`?

Reply:
Apparently most of the "prebuilt"

```
DEBUG: /home/guillaume/.cache/bazel/_bazel_guillaume/9923957c70cd7b59bec21e32af97c8be/external/io_tweag_rules_haskell/haskell/repl.bzl:129:13: <target @io_tweag_rules_haskell_ghc_nixpkgs_haskell_toolchain//:rts, keys:[HaskellInfo, HaskellImportHack, HaskellCoverageInfo, HaskellLibraryInfo, HaddockInfo, OutputGroupInfo]>
DEBUG: /home/guillaume/.cache/bazel/_bazel_guillaume/9923957c70cd7b59bec21e32af97c8be/external/io_tweag_rules_haskell/haskell/repl.bzl:129:13: <target @io_tweag_rules_haskell_ghc_nixpkgs_haskell_toolchain//:ghc-prim, keys:[HaskellInfo, HaskellImportHack, HaskellCoverageInfo, HaskellLibraryInfo, HaddockInfo, OutputGroupInfo]>
DEBUG: /home/guillaume/.cache/bazel/_bazel_guillaume/9923957c70cd7b59bec21e32af97c8be/external/io_tweag_rules_haskell/haskell/repl.bzl:129:13: <target @io_tweag_rules_haskell_ghc_nixpkgs_haskell_toolchain//:integer-gmp, keys:[HaskellInfo, HaskellImportHack, HaskellCoverageInfo, HaskellLibraryInfo, HaddockInfo, OutputGroupInfo]>
DEBUG: /home/guillaume/.cache/bazel/_bazel_guillaume/9923957c70cd7b59bec21e32af97c8be/external/io_tweag_rules_haskell/haskell/repl.bzl:129:13: <target @io_tweag_rules_haskell_ghc_nixpkgs_haskell_toolchain//:base, keys:[HaskellInfo, HaskellImportHack, HaskellCoverageInfo, HaskellLibraryInfo, HaddockInfo, OutputGroupInfo]>
DEBUG: /home/guillaume/.cache/bazel/_bazel_guillaume/9923957c70cd7b59bec21e32af97c8be/external/io_tweag_rules_haskell/haskell/repl.bzl:129:13: <target @io_tweag_rules_haskell_ghc_nixpkgs_haskell_toolchain//:array, keys:[HaskellInfo, HaskellImportHack, HaskellCoverageInfo, HaskellLibraryInfo, HaddockInfo, OutputGroupInfo]>
DEBUG: /home/guillaume/.cache/bazel/_bazel_guillaume/9923957c70cd7b59bec21e32af97c8be/external/io_tweag_rules_haskell/haskell/repl.bzl:129:13: <target @io_tweag_rules_haskell_ghc_nixpkgs_haskell_toolchain//:deepseq, keys:[HaskellInfo, HaskellImportHack, HaskellCoverageInfo, HaskellLibraryInfo, HaddockInfo, OutputGroupInfo]>
DEBUG: /home/guillaume/.cache/bazel/_bazel_guillaume/9923957c70cd7b59bec21e32af97c8be/external/io_tweag_rules_haskell/haskell/repl.bzl:129:13: <target @io_tweag_rules_haskell_ghc_nixpkgs_haskell_toolchain//:bytestring, keys:[HaskellInfo, HaskellImportHack, HaskellCoverageInfo, HaskellLibraryInfo, HaddockInfo, OutputGroupInfo]>
```

Reply:
It was `haskell_import`. I changed the PR so that `haskell_import` now exposes an empty CcInfo.